tests/core/test_pod_security_policies.py

```diff
@@ -60,3 +75,66 @@ def test_service_accounts_have_role_binding(admin_mc, request):
             "default", kubernetes.client.V1DeleteOptions()))
 
     wait_for(lambda: service_account_has_role_binding(rbac, pspt), timeout=30)
+
+
+def test_pod_security_policy_template_del(admin_mc, request, remove_resource):
+    """ Test for pod security policy template binding correctly
+    ref https://github.com/rancher/rancher/issues/15728
+    ref https://localhost:8443/v3/podsecuritypolicytemplates
+    :param admin_mc:
+    :param request:
+    :return:
+    """
+    api_client = admin_mc.client
+
+    setup_cluster_with_pspt(api_client, request)
+    pspt_proj = api_client.create_pod_security_policy_template("pspt2")
+
+    #  create a project in order to establish bindings
+    #  shameless copy from test_rbac.py
+    def create_project():
+        try:
+            p = api_client.create_project(name="test-unrestrict-proj",
+                                          clusterId="local")
+            api_client.wait_success(p)
+            p.setpodsecuritypolicytemplate(
+                podSecurityPolicyTemplateId=pspt_proj.id)
+
+            # In case something goes badly, add a finalizer to
+            # delete the project as the admin
+            remove_resource(p)
+            # Also delete try to delete pspt if we crash out
+            remove_resource(pspt_proj)
+            return p
+        except ApiError as e:
+            assert e.error.status == 403
+            return False
+
+    proj = wait_for(create_project, timeout=30)
+
+    proj = api_client.wait_success(proj)
+    # Create a proj in the local cluster
+    assert proj.state == 'active'
+    assert proj.podSecurityPolicyTemplateId == pspt_proj.id
+
+    # allow for binding deletion
+    api_client.delete(proj)
+
+    def check_project(client, proj):
+        return client.by_id_project(proj.id) is None
+
+    wait_for(lambda: check_project(api_client, proj))
+    # delete the PSPT that was associated with the deleted project
+    api_client.delete(pspt_proj)
+
+    def pspt_exists(client, p):
```

Comment:
the name of this function is "pspt_exists" but it returns True if the pspt _doesnt_ exist. name seems backwards